core/src/main/scala/org/apache/spark/rdd/PipedRDD.scala

```diff
@@ -176,6 +197,8 @@ private[spark] class PipedRDD[T: ClassTag](
 
           false
         }
+        propagateChildThreadException()
```

Comment:
... also this will still not propagate "fatal" exceptions, which includes things like `InterruptedException` and `OutOfMemoryError`. Better to handle `Throwable` for completeness?